command/agent/config/config.go

```diff
@@ -265,6 +276,19 @@ func (c *Config) Merge(c2 *Config) *Config {
 		result.PidFile = c2.PidFile
 	}
 
+	result.Exec = c.Exec
+	if c2.Exec != nil {
+		result.Exec = c2.Exec
+	}
+
+	for _, envTmpl := range c.EnvTemplates {
+		result.EnvTemplates = append(result.EnvTemplates, envTmpl)
+	}
+
+	for _, envTmpl := range c2.EnvTemplates {
+		result.EnvTemplates = append(result.EnvTemplates, envTmpl)
+	}
```

Comment:
@averche changing this to a list removed the ability to override on merge, can you add this to the validation piece (if not part of validation already)

Reply:
I will definitely validate that there are no duplicates in the config.

For the merge behavior, I believe we can still use [slices.ContainsFunc](https://pkg.go.dev/golang.org/x/exp/slices#ContainsFunc) and override if the value exists.

Reply:
eh, would rather not override. if there's dups, just validate and exit

Reply:
In which contexts does merging get used? This should probably dictate whether we override or not.

CC: @VioletHynes

Reply:
Merging is used when e.g. two config files are presented or a directory of config was presented, and we need to merge them to form a single config.

Make sure we follow the same behaviour of how other lists are merged! Here's how Templates are merged (we should use the same strategy, which it looks like we are doing):
```
	for _, l := range c.Templates {
		result.Templates = append(result.Templates, l)
	}
	for _, l := range c2.Templates {
		result.Templates = append(result.Templates, l)
	}

```

By the way, I've been holding off merging until this is flagged ready to review, but if you want earlier review let me know :)

Reply:
yup! that's how I did it. except for regular templates, uniqueness isn't a problem so appending to a list is OK.
for env vars, we are erroring out if there is a duplicate env var in the validation step

Reply:
That makes sense! Validation is I think done (or should be done!) after the merge